pydantic_ai_slim/pydantic_ai/ag_ui.py

```diff
@@ -0,0 +1,871 @@
+    on_startup: Sequence[Callable[[], Any]] | None = None,
+    on_shutdown: Sequence[Callable[[], Any]] | None = None,
+    lifespan: Lifespan[FastAGUI[AgentDepsT, OutputDataT]] | None = None,
+) -> FastAGUI[AgentDepsT, OutputDataT]:
+    """Create a FastAGUI server from an agent.
+
+    Args:
+        agent: The PydanticAI agent to adapt for AG-UI protocol.
+
+        output_type: Custom output type to use for this run, `output_type` may only be used if the agent has
+            no output validators since output validators would expect an argument that matches the agent's
+            output type.
+        model: Optional model to use for this run, required if `model` was not set when creating the agent.
+        deps: Optional dependencies to use for this run.
+        model_settings: Optional settings to use for this model's request.
+        usage_limits: Optional limits on model request count or token usage.
+        usage: Optional usage to start with, useful for resuming a conversation or agents used in tools.
+        infer_name: Whether to try to infer the agent name from the call frame if it's not set.
+        toolsets: Optional list of toolsets to use for this agent, defaults to the agent's toolset.
+
+        debug: Boolean indicating if debug tracebacks should be returned on errors.
+        routes: A list of routes to serve incoming HTTP and WebSocket requests.
+        middleware: A list of middleware to run for every request. A starlette application will always
+            automatically include two middleware classes. `ServerErrorMiddleware` is added as the very
+            outermost middleware, to handle any uncaught errors occurring anywhere in the entire stack.
+            `ExceptionMiddleware` is added as the very innermost middleware, to deal with handled
+            exception cases occurring in the routing or endpoints.
+        exception_handlers: A mapping of either integer status codes, or exception class types onto
+            callables which handle the exceptions. Exception handler callables should be of the form
+            `handler(request, exc) -> response` and may be either standard functions, or async functions.
+        on_startup: A list of callables to run on application startup. Startup handler callables do not
+            take any arguments, and may be either standard functions, or async functions.
+        on_shutdown: A list of callables to run on application shutdown. Shutdown handler callables do
+            not take any arguments, and may be either standard functions, or async functions.
+        lifespan: A lifespan context function, which can be used to perform startup and shutdown tasks.
+            This is a newer style that replaces the `on_startup` and `on_shutdown` handlers. Use one or
+            the other, not both.
+    """
+    adapter: Adapter[AgentDepsT, OutputDataT] = Adapter(agent=agent)
+
+    return FastAGUI(
```

Comment:
Could `Adapter` and `FastAGUI` be merged? Then we could also drop this helper function (and a ton of duplicated args/docstring!)

Reply:
Removed `agent_to_ag_ui`, but think we should still keep Adapter and App separate to allow users to customise how and where its connected, especially now path has been removed for example.